src/System.Text.RegularExpressions/tests/RegexGroupNameTests.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Text.RegularExpressions;
+using System.Collections;
+
+namespace System.Text.RegularExpressionsTests
+{
+    /// <summary>
+    /// Tests the Name property on the Group class.
+    /// </summary>
+    public class RegexGroupNameTests
+    {
+        [Fact]
+        public static void NameTests()
```

Comment:
I think it is worth adding a test for the non-named groups that should end up with names like "0", "1", etc that represent the group number.